python-sdk/src/langwatch/telemetry/span.py

```
@@ -536,115 +579,134 @@ def wrapper(*args: Any, **kwargs: Any):
-                    self._context_token = None
-
-            # We're no longer trying to detach the OpenTelemetry context
-            # since that's what's causing the cross-context issues
-            self._otel_context = None
 
             if self._span is not None:
                 try:
-                    self._span.end()
+                    self._span_context.__exit__(exc_type, exc_value, traceback)
                 except Exception as e:
                     warn(f"Failed to end span: {e}")
```

Ruff (0.8.2)

No explicit `stacklevel` keyword argument found

(B028)